python/cog/server/http.py

```diff
@@ -111,6 +127,86 @@ def healthcheck() -> Any:
             }
         )
 
+    @app.post(
+        "/trainings",
+        response_model=TrainingResponse,
+        response_model_exclude_unset=True,
+    )
+    def train(request: TrainingRequest = Body(default=None), prefer: Union[str, None] = Header(default=None)) -> Any:  # type: ignore
+        """
+        Run a training on the model
+        """
+        if mode != "train":
+            return JSONResponse(
+                {"detail": "The application is running as a predictor, not a trainer."},
+                status_code=400,
+            )
+        if runner.is_busy():
+            return JSONResponse({"detail": "Already training."}, status_code=409)
+
+        respond_async = prefer == "respond-async"
+
+        return _train(request=request, respond_async=respond_async)
+
+    def _train(*, request: TrainingRequest, respond_async: bool) -> Response:
+        # [compat] If no body is supplied, assume that this model can be run
```

Comment:
We might be able to get away with dropping these `[compat]` clauses for trainings.

Reply:
Removed since there's no backwards compatibility to be maintained for this API.